src/internal/util/argsArgArrayOrObject.ts

```diff
@@ -27,6 +27,6 @@ export function argsArgArrayOrObject<T, O extends Record<string, T>>(args: T[] |
   return { args: args as T[], keys: null };
 }
 
-function isPOJO(obj: any): obj is Object {
+function isPOJO(obj: any): obj is {} {
```

Comment:
issue: IMO this should be `object` - note the lower case - rather than `{}`.

`{}` is wider than `object` and will match any value that's neither `null` nor `undefined`. The test within the implementation implies that the result is very much an object.

FWIW, the default `ban-types` configuration in the TypeScript _ESLint_ rule bans `object`, but I disagree with that. TypeScript's own types - e.g. for `WeakMap` - make use of `object`.

Reply:
Given what we're checking here and why we're checking it, maybe `is Record<string, T>` or `is Record<string, any>` is probably what I wanted?

Reply:
`is object` seems to pass, so I'll leave it as you suggest.